### ISSUE TITLE
Is it server less?

### ISSUE THREAD

**Issue by ghost** [martppa/webrtc_react-native#1]

**Comment by martppa** [comment]
Hi! yes, it's peer to peer, you don't need any Webrtc server.

**Comment by ghost** [comment]
could you please let me know when and which time we dont need a server and which time we need server.

I need to add call functionality

**Comment by martppa** [comment]
Normally in a peer to peer Webrtc connection you don't need a server that handles the connection since it's a call between two persons (to devices), for peer to peer you only need a STUN server, in this case hosted by Google (url included in the code) in order to route connections (this server works only as a proxy). In case you want to develop a calling room (more than 2 persons in a call (devices)) then you will have to develop your own server, use a free and open source one, or host your own using any Webrtc hosting provider such as OpenVidu (https://openvidu.io/docs/home/index.html). Hope that helps. :)

Edit: Sorry, OpenVidu is not such a provider. They offer a solution for you to host it. But there other providers that host it for you.

**Comment by ghost** [comment]
Thank you very much for your good information, So do you please tell, Can Call only one person without any server ??

**Comment by ghost** [comment]
I am using android and ios and I am not working with web app

**Comment by martppa** [comment]
Yes, in the example provided in this repository, if you run it on Android (iOS libraries aren't added yet, it's pending) once you run it, it will communicate with itself through this STUN server, it's the same as communication with third device but working on the same device.

**Comment by ghost** [comment]
You mean your example will not on two android device ?

**Comment by martppa** [comment]
I doesn't work on two android devices since it's kinda reflexive, you only have to separate pm and pm2 objects in two different projects and you will get the same result for different devices.

**Comment by ghost** [comment]
Thank you bro, So How can I make it possible to work on two device,  for example I call from one device to other device then he reply and we can speak

**Comment by martppa** [comment]
Hi, sorry for the late reply. Well, I have sadly no time to write the code for you but if you analyse it you will note that there do exist 2 objects called pc and pc2, so they can communicate between themselves. 

Imagine for a moment those objects are in different devices, the flow would be the following:

pc - requests video stream to the device camera
pc2 - requests video stream to the device camera

pc - creates an offer to the STUN server, then the STUN server responses a DESCRIPTION object
pc2 - (in case pc2 is in anothr device you must copy the DESCRIPTION object manually, or use ybou own server (backend app for that) to provide it) pc2 sets the DESCRIPTION object from pc
pc2 - creates an answer to the STUN server, the STUN server responses a DESCRIPTION object for that answer request
pc - set the DESCRIPTION object from pc2 in pc just like otherwise

When the onicecandidate from any of the objects are triggered you must provide it to the other
Whenever the event onaddstream from any of them are triggered, you must then set that stream to the RTCView.

It isn't that  hard, you just have to check the code. :) hope that helps.

**Comment by ghost** [comment]
Thank you very much bro

**Comment by ghost** [comment]
Hello Bro, could you please have a look at this issue?

My code works on android 8.1 real device with local and remote video
But does not work on android 5.0.2 real device with` only remote` video

Source code is here: https://github.com/react-native-webrtc/react-native-webrtc/issues/755
